core/src/main/java/ai/timefold/solver/core/impl/bavet/common/TupleSourceRoot.java

```diff
@@ -0,0 +1,51 @@
+package ai.timefold.solver.core.impl.bavet.common;
+
+import org.jspecify.annotations.NullMarked;
+
+@NullMarked
+public interface TupleSourceRoot<A> {
+    void insert(A a);
```

Comment:
I am not sure why this used to be `@Nullable`, but I remember clearly that I was adding it there for a reason. 

I recommend adding null checks, running the entire CI incl. quickstarts, and if the null checks don't trigger, then I say it's safe for this to be non-null.